src/calliope/backend/backend_model.py

```diff
@@ -1071,13 +1071,14 @@ def _get_variable_bound(
                 name,
                 f"Applying bound according to the {bound} parameter values.",
             )
-            bound_array = self.get_parameter(bound).copy()
+            bound_array = self.get_parameter(bound)
             fill_na = bound_array.attrs.get("default", fill_na)
             references.add(bound)
         else:
             bound_array = xr.DataArray(bound)
-        bound_array.attrs = {}
-        return bound_array.fillna(fill_na)
+        filled_bound_array = bound_array.fillna(fill_na)
+        filled_bound_array.attrs = {}
```

Comment:
I am assuming this empty dict is necessary. Otherwise, i'd go with `del filled_bound_array.attrs`

Reply:
Why prefer `del` over `= {}`?

Reply:
I'm wrong here. It seems like `xarray` automatically puts an empty `dict` in `attrs` by default.
So your approach is correct.
![image](https://github.com/user-attachments/assets/e4c868c4-3919-4216-99e3-1a1752531ec7)